src/ParallelTempering.jl

```diff
@@ -0,0 +1,76 @@
+# vanilla parallel tempering using Julia multithreading
+mutable struct ParallelTempering
+    n_replicas::Int64
+    # temperatures
+    kT_sched::Vector{Float64}
+    # sampler that evolves/updates systems at temperature
+    samplers::Vector{LocalSampler}
+    # system (replica) containing state information 
+    systems::Vector{System}
+    # ids that tells which system is at each sampler
+    system_ids::Vector{Int64}
+    # acceptance rate for replica exchanges btw (rank, rank+1)
+    n_accept::Vector{Int64}
+    # number of attempted exchanges
+    n_exch::Int64
+end
+
+function ParallelTempering(system::System, kT_sched::Vector{Float64}, propose)
+    n_replicas = length(kT_sched)
+    samplers = [LocalSampler(kT=kT, propose=propose) for kT in kT_sched]
```

Comment:
Minor point, but this could be written as `LocalSampler(; kT, propose)` for brevity. Let's do this for now.

Bigger point (for later): The user should have the flexibility to provide their own sampler. They might want Langevin instead of LocalSampler. Perhaps instead of `propose` we should ask the user to provide a sampler and an update function:
```
set_alpha!(sys::System, sampler, alpha::Float64)
```
where alpha will be between 0 and 1. This will support the replica exchange use case, etc.